src/main/java/com/danielfrak/code/keycloak/providers/rest/rest/RestUserService.java

```diff
@@ -2,57 +2,37 @@
 
 import com.danielfrak.code.keycloak.providers.rest.remote.LegacyUser;
 import com.danielfrak.code.keycloak.providers.rest.remote.LegacyUserService;
-import org.jboss.resteasy.client.jaxrs.BasicAuthentication;
-import org.jboss.resteasy.client.jaxrs.ResteasyWebTarget;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.http.HttpStatus;
 import org.keycloak.component.ComponentModel;
 
-import javax.ws.rs.client.Client;
-import javax.ws.rs.core.Response;
+import java.io.IOException;
 import java.util.Optional;
 
 import static com.danielfrak.code.keycloak.providers.rest.ConfigurationProperties.*;
 
 public class RestUserService implements LegacyUserService {
 
-    private final RestUserClient client;
+    private final String uri;
+    private final HttpClient httpClient;
+    private final ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
RE: https://github.com/daniel-frak/keycloak-user-migration/pull/52#discussion_r853152587

If we pass the `objectMapper` in the constructor, we will be able to reuse it in future classes which need it and have more control over our class when unit testing it.